src/utils.py

```diff
@@ -14,7 +16,9 @@ def atomic_write_root_file(path: str, content: bytes, perms: int) -> None:
     :param bytes content: The content to write to the file.
     :param int perms: The file permissions to set on the created file.
     """
-    logger.debug("Writing file {} with root ownership and permissions {:o}".format(path, perms))
+    logger.debug(  # pylint: disable=logging-fstring-interpolation
+        f"Writing file {path} with root ownership and permissions {perms:o}"
```

Comment:
It's not a good practice to use f-string for logging.
Please update to use the lazy format using `%s`

Reply:
Thanks, updated. :)